stock_landed_costs_average/tests/test_stock_landed_costs_average.py

```diff
@@ -0,0 +1,179 @@
+            'Something went wrong. Invoice should be paid!!!')
+
+        wizard_obj = self.env['attach.invoice.to.landed.costs.wizard']
+        wizard_brw = wizard_obj.with_context(
+            {'active_id': invoice_id}).create(
+                {'stock_landed_cost_id': landed_id})
+        wizard_brw.add_landed_costs()
+        self.assertEquals(
+            bool(invoice_brw.stock_landed_cost_id.id), True,
+            'Invoice should be attached to a Landed Cost!!!')
+        return True
+
+    def revert_landed_cost(self, landed_cost_id):
+        reverted_landed_cost = landed_cost_id.copy()
+        reverted_landed_cost.write({
+            'picking_ids': [(4, self.picking_id.id)],
+        })
+
+        for line in reverted_landed_cost.cost_lines:
+            line.write({'price_unit': -1 * line.price_unit})
+        reverted_landed_cost.compute_landed_cost()
+        reverted_landed_cost.button_validate()
+        self.assertEquals(
+            reverted_landed_cost.state, 'done',
+            'Something went wrong. Reverted Landed Costs should be Posted!!!')
+        self.reverted_landed_cost = reverted_landed_cost
+        return True
+
+    def create_landed_cost(self):
+        self.create_picking()
+        picking_id = self.picking_id
+        self.landed_cost_id = self.env['stock.landed.cost'].create({
+            'picking_ids': [(4, picking_id.id)],
+            'account_journal_id': self.ref('account.expenses_journal'),
+        })
+
+        landed_cost_id = self.landed_cost_id
+        self.assign_invoice(landed_cost_id.id)
+
+        landed_cost_id.get_costs_from_invoices()
+        landed_cost_id.compute_landed_cost()
+        landed_cost_id.button_validate()
+        self.assertEquals(
+            landed_cost_id.state, 'done',
+            'Something went wrong. Landed Costs should be Posted!!!')
+
+        product_average = self.product_average
+        product_real = self.product_real
+        self.assertEquals(
+            product_average.standard_price, 300.0,
+            'Something went wrong. Average Product should cost 300.00!!!')
+        self.assertEquals(
+            product_real.standard_price, 0.0,
+            'Something went wrong. Real Product should cost 0.00!!!')
+        quant = self.env['stock.quant']
+        quant_real = quant.search([('product_id', '=', product_real.id)])
+        quant_average = quant.search([('product_id', '=', product_average.id)])
+
+        self.assertEquals(
+            (quant_average.inventory_value, quant_average.cost),
+            (3000.0, 300.0),
+            'Something went wrong. Average Product value is 3000.00!!!')
+
+        self.assertEquals(
+            (quant_real.inventory_value, quant_real.cost),
+            (3000.0, 300.0),
+            'Something went wrong. Real Product value is 3000.00!!!')
+
+        return True
+
+    def test_basic_landed(self):
+        self.create_landed_cost()
+        self.revert_landed_cost(self.landed_cost_id)
```

Comment:
@nhomar 

Landed Cost Reversion will fail till https://github.com/odoo/odoo/pull/8714
is not solved by odoo.